### PR TITLE
feat(x/ecocredit): add create-basket command

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #732

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
Updates pending #757

File: x/ecocredit/client/basket/tx_create.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	cmd.Flags().String(FlagMinimumStartDate, "", "the earliest start date for batches of credits allowed into the basket (e.g. \"2012-01-01\")")
+	cmd.Flags().String(FlagStartDateWindow, "", "sets a cutoff for batch start dates when adding new credits to the basket (e.g. \"1325404800\")")
+	cmd.Flags().String(FlagBasketFee, "", "the fee that the curator will pay to create the basket (e.g. \"20regen\")")
+
+	// required flags
+	cmd.MarkFlagRequired(FlagDisplayName)
+	cmd.MarkFlagRequired(FlagExponent)
+	cmd.MarkFlagRequired(FlagCreditTypeName)
+	cmd.MarkFlagRequired(FlagAllowedClasses)
+	cmd.MarkFlagRequired(FlagBasketFee)
```
Comment (technicallyty) [inline x/ecocredit/client/basket/tx_create.go:202]: is fee actually required? iirc there was a discussion around empty fees in the validate basic for MsgCreate being ok

**Review by technicallyty — APPROVED** [review]
lgtm - one question

File: x/ecocredit/client/basket/tx_create.go (hunk 1 above)
Comment (aaronc) [reply x/ecocredit/client/basket/tx_create.go:202]: fee is not required

File: x/ecocredit/client/basket/tx_create.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	cmd := &cobra.Command{
+		Use:   "create-basket [name]",
+		Short: "Creates a bank denom that wraps credits",
+		Long: strings.TrimSpace(`Creates a bank denom that wraps credits
+
+Parameters:
+		name: the name used to create a bank denom for this basket token.
+
+Flags:
+		display_name: the name used to create a bank denom display name.
```
Comment (aaronc) [inline x/ecocredit/client/basket/tx_create.go:41]: Can't wait till lens is fully ready and we don't need to write CLI commands anymore and copy all the docs in the .proto files manually lol

**Review by aaronc — APPROVED** [review]
Preapproving

**Pull request merged**